### PR TITLE
Release notes improvements

### PR DESCRIPTION
**Pull request by emartynov**
## Notes
* Please label this PR with the `enhancement` label

## Changes
* Introduced a [github action](https://github.com/mikepenz/release-changelog-builder-action) to form release notes based on the PR labels
  * feature, enhancement; fix, bug; maintenance (see [configuration](.gitlab/changelog-configuration.json))
  * [example](https://github.com/mikepenz/release-changelog-builder-action/releases)
* Replaced script with a [github action](https://github.com/softprops/action-gh-release) to create the release
* Deleted the old variant

## Screenshots
The look should be identical:
| Original | New |
|--------|-------|
| Screenshot 2021-03-27 at 23 28 08 | Screenshot 2021-03-27 at 23 39 05 |

## Proposal
* Do not run build for all variants on the release to decrease the build time

### CODE REVIEW

File: .github/workflows/release-changelog.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: Changelog vs Release
+
+on:
+  push:
+    tags:
+      - 'v*'
```
Comment (hannesa2) [inline .github/workflows/release-changelog.yml:6]: Why only starting with a `v` ?
To be honest, I see no point to request it

File: .github/workflows/release-changelog.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+jobs:
+  release:
+    if: startsWith(github.ref, 'refs/tags/')
```
Comment (hannesa2) [inline .github/workflows/release-changelog.yml:10]: For me it looks like a redundant condition. 

It's only triggered by a tag https://github.com/MikeOrtiz/TouchImageView/pull/337/files#diff-305377beb31aab4ccc3ffbe29a11cd428b5a52460bc855ceba49c92faec46bf3R5

File: .github/workflows/release-changelog.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (hannesa2) [inline .github/workflows/release-changelog.yml:14]: You simplified too much
* There is already a newer action
* You use only a shallow clone. The full history is used to generated versioncode from commit count

Please go with previous checkout

**Review by hannesa2 — CHANGES_REQUESTED** [review]
Cool, thank you !

Some little things: I would love to use old action name `Android-CI-release.yml` (If you like the name or not). Then we all see you changes better.

We (or I) can rename it later too.

Btw, is there an action to label it anyhow automatic ? 
It's not requested now, but interesting to know

File: .github/workflows/release-changelog.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        uses: malinskiy/action-android/install-sdk@release/0.1.0
+
+      - name: Build project
+        run: ./gradlew clean build
+        env:
+          VERSION: ${{ github.ref }}
+
+      - name: Get the version
+        id: get_version
+        run: echo ::set-output name=VERSION::${GITHUB_REF#refs/tags/}
```
Comment (hannesa2) [inline .github/workflows/release-changelog.yml:39]: What's about to use this ?

      - name: Get the version
        id: tagger
        uses: jimschubert/query-tag-action@v2
        with:
          skip-unshallow: 'true'
          abbrev: false
          commit-ish: HEAD

and use it with `${{steps.tagger.outputs.tag}}`

File: .github/workflows/release-changelog.yml (hunk 1 above)
Comment (emartynov) [reply .github/workflows/release-changelog.yml:6]: Done

File: .github/workflows/release-changelog.yml (hunk 2 above)
Comment (emartynov) [reply .github/workflows/release-changelog.yml:10]: Removed

File: .github/workflows/release-changelog.yml (hunk 3 above)
Comment (emartynov) [reply .github/workflows/release-changelog.yml:14]: Fixed

File: .github/workflows/release-changelog.yml (hunk 4 above)
Comment (emartynov) [reply .github/workflows/release-changelog.yml:39]: Up to you. Propose to do it in different PR - don't want to lose time with testing. If you don't mind.

Comment (emartynov) [reply .github/workflows/release-changelog.yml:39]: Ah wait, it is just easier for me to test it now. Let me try it.

File: .github/workflows/Android-CI-release.yml [hunk 5]
```diff
@@ -1,34 +1,54 @@
...
+        with:
+          skip-unshallow: 'true'
+          abbrev: false
+          commit-ish: HEAD
+
+      - name: Create Release
+        uses: softprops/action-gh-release@v1
+        with:
+          tag_name: ${{ github.ref }}
+          name: ${{ steps.get_version.outputs.VERSION }}
```
Comment (hannesa2) [inline .github/workflows/Android-CI-release.yml:50]: Almost done, please use 

`name: ${{steps.tagger.outputs.tag}}`

File: .github/workflows/release-changelog.yml (hunk 4 above)
Comment (hannesa2) [reply .github/workflows/release-changelog.yml]: This is currently missing

https://github.com/MikeOrtiz/TouchImageView/pull/337#discussion_r602863105

File: .github/workflows/Android-CI-release.yml (hunk 5 above)
Comment (emartynov) [reply .github/workflows/Android-CI-release.yml:50]: Missed, doing it

**Review by hannesa2 — APPROVED** [review]
Thank you !

**Comment by emartynov** [comment]
And yes, some extent of the automatic labelling is possible also https://mobilediana.medium.com/adding-labels-automatically-using-github-actions-c0f54f540715

**Comment by hannesa2** [comment]
I squashed your changes and release is running https://github.com/MikeOrtiz/TouchImageView/actions/runs/694928999

I'm excited about the outcome

**Comment by emartynov** [comment]
My local actions just finished and the result looks the same https://github.com/emartynov/TouchImageView/releases

**Comment by hannesa2** [comment]
Unfortunately my squash destroy the changelog about this change.
I should force this by the pull request creator, but I want to avoid to annoy to much

**Comment by hannesa2** [comment]
How to suppress eg the dependabot updates ?

**Comment by emartynov** [comment]
Revert it and I still have a branch - so I can create new PR. I will sync the upstream before.

**Comment by hannesa2** [comment]
Ok, I will revert .... one moment

**Comment by emartynov** [comment]
Remove the label from the config file - and dependencies will not be in the CHANGELOG. Same about Gradle updates label.

**Comment by hannesa2** [comment]
I reverted it, please recreate your pull request

**Comment by hannesa2** [comment]
and please quash the commits

**Comment by emartynov** [comment]
Which commits should I squash and why?

**Comment by hannesa2** [comment]
Two files with 16 commits is definitive too much.
Please squash all

**Comment by emartynov** [comment]
You can do it with GitHub on merge, correct?

**Comment by hannesa2** [comment]
Of course, I can do this as well

**Pull request merged**